nav2_waypoint_follower/plugins/input_at_waypoint.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2020 Samsung Research America
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <pluginlib/class_list_macros.hpp>
+#include <string>
+#include <exception>
+
+#include "nav2_util/node_utils.hpp"
+#include "nav2_waypoint_follower/plugins/wait_at_waypoint.hpp"
+
+namespace nav2_waypoint_follower
+{
+
+using std::placeholders::_1;
+
+InputAtWaypoint::InputAtWaypoint()
+: input_received_(false),
+  is_enabled_(true)
+{
+}
+
+InputAtWaypoint::~InputAtWaypoint()
+{
+}
+
+void InputAtWaypoint::initialize(
+  const rclcpp_lifecycle::LifecycleNode::WeakPtr & parent,
+  const std::string & plugin_name)
+{
+  auto node = parent.lock();
+  
+  if (!node) {
+    throw std::runtime_error{"Failed to lock node in input at waypoint plugin!"};
+  }
+
+  logger_ = node->get_logger();
+  clock_ = node->get_clock();
+
+  nav2_util::declare_parameter_if_not_declared(
+    node, plugin_name + ".timeout",
+    rclcpp::ParameterValue(10));
+  nav2_util::declare_parameter_if_not_declared(
+    node, plugin_name + ".enabled",
+    rclcpp::ParameterValue(true));
+  node->get_parameter(plugin_name + ".timeout", timeout);
```

Comment:
something with `timeout` seems not right

Reply:
What do you mean?

Reply:
forgot what exactly I meant here too but I guess I meant `timeout` should be `timeout_` here

Reply:
It needs to be timeout as a local variable so I can get the `double` type from input to convert it to a `rclcpp::Duration` type below